droidconKE/chai
# [Feature] Asset Provider

name: Asset Provider
about: A crazy first approach for accessible assets 😅 
label: 'WIP'

---

## Description
This is just an idea to handle assets in a more accessible way. There are a few things that I still don't like, such as passing the android context around. I did limit this as best as possible and would work to eliminate it altogether so that one can access the assets without it. `ChaiImage.DCKE_MAIN` vs. `ChaiImage.DCKE_MAIN(ctx)`.

⚠️ I didn't want to spend too much more time on this if this isn't a viable solution or something we want to pursue further.

Anyway, let me know what you think 🙌 Let's discuss this one 🤓 

---
## How can this PR be tested?
You can test it in a few ways:
* Run the tests 🧪 
* Run ChaiDemo, here you can see the idea in action 🚀 

---

Thank you for opening a Pull Request!
Before submitting your PR, there are a few things you can do to make sure it goes smoothly:


Is this your first Pull Request?

Fixes #<issue_number_goes_here> 🦕

# Screenshot
![Screenshot_20230324_204158](https://user-images.githubusercontent.com/7081242/227624144-9dbd842b-6524-43a4-85fd-11f22af17bd8.png)
